boot/loongarch/startup64.S

```diff
@@ -0,0 +1,465 @@
+// SPDX-License-Identifier: GPL-2.0
+//
+// startup64.S contains the 64-bit startup code for both the BSP and APs.
+// It initialises stacks, memory management, and exception handling, clears
+// the BSS, completes relocation, and finally calls the main application.
+//
+// Copyright (C) 2020-2022 Martin Whitaker.
+//
+// Derived from memtest86+ head.S:
+//
+// linux/boot/head.S
+// Copyright (C) 1991, 1992  Linus Torvalds
+// 1-Jan-96 Modified by Chris Brady for use as a boot/loader for MemTest-86.
+// Set up the memory management for flat non-paged linear addressing.
+// 17 May 2004 : Added X86_PWRCAP for AMD64 (Memtest86+ - Samuel D.)
+
+#define __ASSEMBLY__
+
+#include "boot.h"
+#inlucde "interrupt.h"
+
+#define RSIZE                 8           // 64 bit mode register size
+#define GP_REG_CONTEXT_SIZE   32 * RSIZE  // General-purpose registers size
+#define FP_REG_CONTEXT_SIZE   34 * RSIZE  // Floating-point registers size
+#define CSR_REG_CONTEXT_SIZE  9  * RSIZE  // CSR registers size
+
+#define NUM_INT_VEC	20
+
+	.text
+
+exception_entry:
+  csrwr   $sp, 0x30
+
+  csrrd   $sp, 0x30
```

Comment:
> It could be helpful to `#define` the CSR numbers so the usage sites are much more self-descriptive. For example here `0x30` is `SAVE0` so it's immediately obvious that we're stashing the original SP into a scratch CSR.

This code is very samplified version of EDK2 exception handler, so some definitions are not sync with EDK2 version. I think we can defined them if it is necessary.
> 
> And BTW why is it necessary to read back the SP immediately after the write above?

Because the stack layout has to do this.